contentcuration/contentcuration/frontend/channelEdit/views/files/FileUploadItem.vue

```diff
@@ -53,15 +45,20 @@
           </VListTileContent>
           <VSpacer />
           <VListTileAction v-if="fileDisplay">
-            <div v-if="allowFileRemove" class="remove-icon">
-              <IconButton
-                icon="clear"
-                color="grey"
-                :text="$tr('removeFileButton')"
-                data-test="remove"
-                @click="$emit('remove', file)"
-              />
-            </div>
+            <KIconButton
+              size="small"
+              icon="optionsHorizontal"
+              appearance="flat-button"
+              data-test="show-file-options"
+            >
+              <template #menu>
+                <KDropdownMenu
+                  :options="previewFilesOptions"
+                  data-test="file-options"
```

Comment:
You are passing `openFileDialog` as an argument to every option's `onClick` handler. But in the`previewFilesOptions`, only the 'REPLACE_FILE' option actually expects that argument?  The download and remove options don't need it.  I think this can cause other handlers to receive `openFileDialog` as an unexpected parameter

I assumed when the download option is clicked, `this.downloadFile()` is called with `openFileDialog` as an argument. If `downloadFile` isn't written to handle parameters same applies to `removeFile`.

Reply:
Thanks for raising this Allan! I'll make some changes to have the handlers arguments more explicit

Reply:
@AllanOXDi, on thinking about this some more, I actually think the current approach should be fine too. We are currently passing and invoking the function argument in the replace file `onClick` method only. The rest have no argument definitions, so JS simply ignores them even if they are passed, so it shouldn't have any ramifications as far as I can tell.